### PR TITLE
[MIG]-Migrated Business requirement deliverable cost module from 8.0 t…

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
…o 10.0
#48 
- This PR depends on #223 and #224.
- Migrated Business_requirement_deliverable_cost from v8 to v10.

### CODE REVIEW

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS  Can you add the relevant 2 printouts from #232?

**Comment by elicoidal** [comment]
#224 and #223 are merged: please rebase

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
Rebased

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS my feedback:
- [X] If the user is not in estimation group he should not see the total of deliverable in the smart button for deliverables #248 
- [x] if user belongs to cost group he should see the total of the resources in the smart button for resources.
- [X] Demo product has no cost set. SER01 should be 25, SER02->50, SER03->30, WEB01->100, WEB02->3639. #248 
- [X] in deliverable and in BR, if gross profit is negative, it should be in red (is it possible?). CANCELED

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
>Demo product has no cost set. SER01 should be 25, SER02->50, SER03->30, WEB01->100, WEB02-3639.

I think cost should be different for SER02, WEB02
I have checked in v8 for SER02 having cost value 35, and WEB02 having 3500.
please correct me if I am wrong.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS thanks for checking it: please follow v8 costs :smile:

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
Some Of the changes related BRD module so I have made another PR:
Could you please refer it #248

**Comment by elicoidal** [comment]
>  in deliverable and in BR, if gross profit is negative, it should be in red (is it possible?)

Forget about it

**Review by elicoidal — APPROVED** [review]
cc @victormartinelicocorp @seb-elico

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
For Business Cost control group in selection field, I have made changes here.
Once runbot is ok you can test here.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS this is still the old behavior for the groups in the user. Can you clarify what needs to be checked in this PR?

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
for user groups I have made changes on BR PR #248

**Comment by elicoidal** [comment]
in https://github.com/OCA/business-requirement/pull/248#issuecomment-330219003 you say:
> Cost control group as selection field you can test here #225

But here the cost control group is not in selection field: I am a little bit lost

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
I have checked in runbot.
cost control group in selection field is there.

peek 2017-09-19 15-22test

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS True: I got confused by the other group which is still in list below.
I assume it will be fixed when rebasing/merging.

**Comment by elicoidal** [comment]
@victormartinelicocorp Can you have a look?

**Review by martintorresvm — APPROVED** [review]
The diff seems ok to me. 
Need to be tested

**Comment by YogeshMahera-SerpentCS** [comment]
Fixed issues of #253

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Can you please review it!

**Review by martintorresvm — APPROVED** [review]
To be tested

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS this is not satisfactory

- [x] Currency in DL should be the one from the Pricelist.
- [x] Cost control fields in DL and BR should follow the format: 
  * Total Revenue / Total Revenue (Cie Curr.) / Total Procurement (Cie Curr.)/ Total Tasks (Cie Curr.)/Est. Gross Profit (Cie Curr.)
  * first field in the BR Pricelist currency / all other fields in the Company Currency
- [x] Currency should be displayed in all lines (like you had in a previous version

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp @elicoidal Can you please review it

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp runbot is green

**Review by martintorresvm — APPROVED** [review]
To be tested

**Comment by elicoidal** [comment]
Small details left: overall much better
- [ ] in general adding "(Cie Curr.)" makes it ugly in 2 lines: in cost control and deliverable, can you make the title 2 column wide? If not possible then we remove the text "(Cie Curr.)" (it is now self explanatory with the currency symbol)

- [ ] If PL currency = Company currency, best is not display the "Total revenue (cie curr.)"

File: business_requirement_deliverable_cost/models/business.py [hunk 1]
```diff
@@ -134,6 +134,21 @@ def _default_currency(self):
         default=_default_currency,
         help="Company Currency Id"
     )
+    currency_status = fields.Boolean(
+        compute='get_currency',
+        string='Company Currency'
+    )
+
+    @api.depends('business_requirement_id.pricelist_id')
+    def get_currency(self):
```
Comment (martintorresvm) [inline business_requirement_deliverable_cost/models/business.py]: @YogeshMahera-SerpentCS please follow OCA guidelines.

s/get_currency/compute_currency_status/

**Review by martintorresvm — CHANGES_REQUESTED** [review]

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp  Could you please review it

**Comment by YogeshMahera-SerpentCS** [comment]
> in general adding "(Cie Curr.)" makes it ugly in 2 lines: in cost control and deliverable, can you make the title 2 column wide? If not possible then we remove the text "(Cie Curr.)" (it is now self explanatory with the currency symbol)

I have tried this but not showing properly so removed this (Cie Curr.)

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS 

>  If PL currency = Company currency, best is not display the "Total revenue (cie curr.)"

Is it possible to do?

- [x] in DL: hide the field Currency (between Sales price and Total Revenue)
- [x] Rename "Total revenue" into "Total Deliverable" (or Deliver. if it doesnot fits in the column) in both Deliverable and BR screen 
- [x] Rename "Total revenue (cie curr.)"  into "Total Revenue"  in both Deliverable and BR screen  
- [x] the field (initially) "Total revenue (cie curr.)" in BR/cost control is in strange place and should be instead of "Total revenue" (exchange place between both fields)

@YogeshMahera-SerpentCS

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal runbot is green

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS Normally last small detail to be fixed:
- [x] Pricelist can be modified only in draft status.

Normally fixed somewhere else:
- [ ] if not group estimation: total task should not appear in BR/Cost control (nor the cost control tab should appear actually) and not the Total procurement/Total Tasks/Est. Gross profit in the Deliverable form. I think it is linked to rebase the module for the cost/estimation inherit group
- [ ] Deliverable Total should not appear if not estimation group (must be fixed in another PR)

Separate small issues:
- [x] in log tab: Creation date is not aligned properly
screenshot from 2017-09-29 10-15-19

- [x] in other: there is a line in the bottom right which should not appear
screenshot from 2017-09-29 10-14-54

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal please  check

**Review by elicoidal — APPROVED** [review]
All good!

**Comment by elicoidal** [comment]
@victormartinelicocorp I move forward as your concerns have been attended

**Pull request merged**